lib/main.dart

```diff
@@ -1,113 +1,135 @@
-    // fast, so that you can just rebuild anything that needs updating rather
-    // than having to individually change instances of widgets.
     return Scaffold(
-      appBar: AppBar(
-        // Here we take the value from the MyHomePage object that was created by
-        // the App.build method, and use it to set our appbar title.
-        title: Text(widget.title),
-      ),
-      body: Center(
-        // Center is a layout widget. It takes a single child and positions it
-        // in the middle of the parent.
-        child: Column(
-          // Column is also a layout widget. It takes a list of children and
-          // arranges them vertically. By default, it sizes itself to fit its
-          // children horizontally, and tries to be as tall as its parent.
-          //
-          // Invoke "debug painting" (press "p" in the console, choose the
-          // "Toggle Debug Paint" action from the Flutter Inspector in Android
-          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
-          // to see the wireframe for each widget.
-          //
-          // Column has various properties to control how it sizes itself and
-          // how it positions its children. Here we use mainAxisAlignment to
-          // center the children vertically; the main axis here is the vertical
-          // axis because Columns are vertical (the cross axis would be
-          // horizontal).
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            Text(
-              'You have pushed the button this many times:',
+        appBar: AppBar(
+          title: Text('Innopolis Feedback'),
+        ),
+        body: Column(
+          children: [
+            Padding(
+              padding: const EdgeInsets.all(15.0),
+              child: Text('Welcome to Innopolis Feedback!',
+                  style: TextStyle(fontSize: 22)),
             ),
-            Text(
-              '$_counter',
-              style: Theme.of(context).textTheme.headline4,
+            Expanded(
+              child: FutureBuilder(
+                future: getYears(),
+                builder: (context, snapshot) {
+                  if (snapshot.hasData) {
+                    years = snapshot.data;
+                    if (selectedYear == null) {
+                      currentList = years;
+                      currentBuilder = yearItemBuilder;
+                    }
+                    return ListView.builder(
+                        padding: const EdgeInsets.all(8),
+                        itemCount: currentList.length,
+                        itemBuilder: (context, index) =>
+                            currentBuilder(currentList[index]));
+                  }
+                  if (snapshot.hasError)
+                    return Text('Oops! Something went wrong :(');
+                  return Text('Loading ...');
+                },
+              ),
```

Comment:
Am I right, that first-time Future builder and then just ListViews because initially, we need to get all the info from some external place and then just iterate over it?

Reply:
Yes, you can say so, I guess.